windows/RNLocalize/RNLocalizeModule.cpp

```diff
@@ -0,0 +1,190 @@
+#include "pch.h"
+
+#include "RNLocalizeModule.h"
+
+#include "winrt/Windows.Globalization.h"
+
+using namespace winrt::Microsoft::ReactNative;
+
+namespace winrt::RNLocalize
+{
+
+std::vector<winrt::hstring> RNLocalizeModule::getLocaleNames()
+{
+    std::vector<winrt::hstring> locales;
+    for (winrt::hstring locale : winrt::Windows::Globalization::ApplicationLanguages::Languages())
+    {
+        locales.push_back(locale);
+    }
+    return locales;
+}
+
+std::vector<Locale> RNLocalizeModule::getLocales(std::vector<winrt::hstring> localeStrings)
+{
+    std::vector<Locale> localesArray;
+    for (winrt::hstring localeString : localeStrings)
+    {
+        Locale locale;
+        std::string languageCode = getLanguageCode(to_string(localeString));
+        locale.languageCode = languageCode;
+
+        std::string countryCode = getCountryCode(to_string(localeString));
+        locale.countryCode = countryCode;
+
+        locale.isRTL = getIsRTL(languageCode);
+
+        std::string scriptCode = getScriptCode(localeString);
+        locale.scriptCode = scriptCode;
+
+        std::string languageTag = languageCode + "-" + scriptCode + "-" + countryCode;
+        locale.languageTag = languageTag;
+
+        localesArray.push_back(locale);
+    }
+    return localesArray;
+}
+
+std::string RNLocalizeModule::getCountryCode(std::string locale)
+{
+    return locale.substr(locale.find("-") + 1, locale.length());
```

Comment:
this will be wrong for some locales, e.g.:

Bosnian (Cyrillic, Bosnia and Herzegovina)  bs-Cyrl-BA

See https://docs.microsoft.com/en-us/windows/uwp/publish/supported-languages

I think you might want to use the actual API instead of parsing the locale code: https://docs.microsoft.com/en-us/uwp/api/windows.globalization.geographicregion?view=winrt-19041

Reply:
Switched to new API!